src/messages/textDocument_rename.cc

```diff
@@ -21,12 +21,18 @@ namespace {
 WorkspaceEdit BuildWorkspaceEdit(DB *db, WorkingFiles *wfiles, SymbolRef sym,
                                  const std::string &new_text) {
   std::unordered_map<int, TextDocumentEdit> path_to_edit;
+  Location last_location; // prevent duplicates
 
   EachOccurrence(db, sym, true, [&](Use use) {
     std::optional<Location> ls_location = GetLsLocation(db, wfiles, use);
     if (!ls_location)
       return;
 
+    if (*ls_location == last_location)
```

Comment:
Thanks for reminding me of this. I think a more robust approach is to check if the string literal exists in the replacement sites. I'll take a stab.

Reply:
This approach works fine for my case. String literal exists in replacement site, but the same range is mentioned twicely. Can you describe your idea?